Framework/PythonInterface/test/python/plugins/algorithms/SaveP2DTest.py

```diff
@@ -0,0 +1,96 @@
+         with open(os.path.join(self._test_dir, "ref_savep2d_remove_NaN.txt", 'w')) as of:
+            of.write(text)
+         self._assert_file_content_equal(os.path.join(self._test_dir, "ref_savep2d_remove_NaN.txt"), os.path.join(self._test_dir, "test_savep2d_remove_NaN.p2d"))
+
+    def test_savep2d_cut_data(self):
+         SaveP2D(Workspace = self._workspace, OutputFile = os.path.join(self._test_dir, "test_savep2d_cut_data"), RemoveNaN = False, RemoveNegatives = False, CutData = True, tthMin = 10, tthMax = 140, lambdaMin = 1.4, lambdaMax = 2.496, dMin = 2.5, dMax = 7.5, dpMin = 1, dpMax = 3)
+         text = '#Title: test\n#Inst: .prm\n#Binning: ddperp   0.8888889    1.0000000\n#Bank: 1\n#2theta   lambda   d-value   dp-value   counts\n  28.5401669      1.4789581      3.0000000      1.5000000      3.0000000\n  21.4420009      1.4882141      4.0000000      1.5000000     -1.0000000\n  17.1666094      1.4924723      5.0000000      1.5000000      0.0000000\n  14.3112545      1.4947782      6.0000000      1.5000000            nan\n  12.2697184      1.4961662      7.0000000      1.5000000      3.0000000\n  48.4687709      2.4628222      3.0000000      2.5000000      3.0000000\n  36.1141714      2.4797153      4.0000000      2.5000000     -1.0000000\n  28.8035116      2.4871957      5.0000000      2.5000000      0.0000000\n  23.9632304      2.4911738      6.0000000      2.5000000            nan\n  20.5194188      2.4935442      7.0000000      2.5000000      3.0000000\n'
+         with open(os.path.join(self._test_dir, "ref_savep2d_cut_data.txt", 'w')) as of:
+            of.write(text)
+         self._assert_file_content_equal(os.path.join(self._test_dir, "ref_savep2d_cut_data.txt"), os.path.join(self._test_dir, "test_savep2d_cut_data.p2d"))
+
+    def test_savep2d_cut_data_dp(self):
+         SaveP2D(Workspace = self._workspace, OutputFile = os.path.join(self._test_dir, "test_savep2d_cut_data_dp"), RemoveNaN = False, RemoveNegatives = False, CutData = True, tthMin = 10, tthMax = 140, lambdaMin = 1.4, lambdaMax = 2.496, dMin = 2.5, dMax = 7.5)
+         text = '#Title: test\n#Inst: .prm\n#Binning: ddperp   0.8888889    1.0000000\n#Bank: 1\n#2theta   lambda   d-value   dp-value   counts\n  28.5401669      1.4789581      3.0000000      1.5000000      3.0000000\n  21.4420009      1.4882141      4.0000000      1.5000000     -1.0000000\n  17.1666094      1.4924723      5.0000000      1.5000000      0.0000000\n  14.3112545      1.4947782      6.0000000      1.5000000            nan\n  12.2697184      1.4961662      7.0000000      1.5000000      3.0000000\n'
+         with open(os.path.join(self._test_dir, "ref_savep2d_cut_data_dp.txt", 'w')) as of:
```

Comment:
These brackets are misalligned, the 'w' flag is being passed into `os.path.join` rather than `open`
```suggestion
         with open(os.path.join(self._test_dir, "ref_savep2d_cut_data_dp.txt"), 'w') as of:
```